jina/helper.py

```diff
@@ -1322,7 +1322,10 @@ def run(self):
                 'please report this issue here: https://github.com/jina-ai/jina'
             )
     else:
-        return get_or_reuse_loop().run_until_complete(func(*args, **kwargs))
+        loop = get_or_reuse_loop()
+        result = loop.run_until_complete(func(*args, **kwargs))
+        loop.close()
```

Comment:
We can only close a loop, if we created it. If it was already running, we shouldn't.

Reply:
we should use some context manager here

Reply:
we could, its a bit tricky though. The underlying design issue is that the caller does not necessarily own the loop.

Reply:
I implemented now in the helper and used [here](https://github.com/jina-ai/jina/pull/4734/files#diff-91bf316ed025400aa871f71855a824d46198a819fe561abae015373e9cbb7f65R1346)